src/lib.rs

```diff
@@ -1034,6 +1035,174 @@ impl FromStr for Cultures {
         }
     }
 }
+/// A PathBuf that will be in the output of print (and therefore saved to disk)
+///
+/// A proper PathBuf should not be used for that, as we don't want to introduce
+/// platform-specific separators.
+///
+/// This type intentionally lacks some path functionality like `.exists()` because
+/// we don't want to be using it for *actual* path stuff, only for writing it to output.
+/// Most StoredPathBufs are just user-provided strings with no processing applied.
+///
+/// However sometimes we are forced to handle a PathBuf because of things like
+/// cargo-metadata, in which case [`StoredPathBuf::from_utf8_path`][] or
+/// [`StoredPathBuf::from_std_path`][] will convert the path to the windows path style.
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct StoredPathBuf(String);
+
+/// A borrowed [`StoredPathBuf`][]
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct StoredPath(str);
+
+impl StoredPathBuf {
+    /// Make a new StoredPathBuf from a String
+    pub fn new(v: String) -> Self {
+        Self(v)
+    }
+
+    /// Make a new StoredPath from a OS-specific path
+    ///
+    /// This breaks the path into its components and rewrites the slashes to `\`.
+    ///
+    /// Generally you should avoid this and just try to preserve the user input,
+    /// but it's required when using things like the output of cargo-metadata.
+    pub fn from_std_path(path: &Path) -> Option<Self> {
+        Utf8Path::from_path(path).map(Self::from_utf8_path)
+    }
+
+    /// Make a new StoredPath from a OS-specific Utf8Path
+    ///
+    /// This breaks the path into its components and rewrites the slashes to `\`.
+    ///
+    /// Generally you should avoid this and just try to preserve the user input,
+    /// but it's required when using things like the output of cargo-metadata.
+    ///
+    /// Also note that this does some handling of absolute paths, but that those
+    /// are kind of nonsensical to store longterm. Still, the user can hand them
+    /// to us, and we have to do our best to deal with it. Mostly this just comes
+    /// up in test code.
+    pub fn from_utf8_path(path: &Utf8Path) -> Self {
```

Comment:
Are [UNC paths](https://users.rust-lang.org/t/understanding-windows-paths/58583) going to be an issue? Would [dunce](https://gitlab.com/kornelski/dunce) be useful here?

Reply:
In general the implications of my implementation are:

* If we're on windows, nothing has really changed
* If we're on unix, std::Path and std::Utf8Path (mostly stuff from cargo_metadata) will aggressively have `/` converted to `\\`

Reply:
err, which is to say: unc paths shouldn't be affected, as a strictly windows thing